### PR TITLE
Properly cancel event when beforeunload handler returns null

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15142)

### CODE REVIEW

File: components/script/dom/eventtarget.rs [hunk 1]
```diff
@@ -179,22 +179,15 @@ impl CompiledEventListener {
...
-                                                             exception_handle) {
-                                match value {
-                                    Some(value) => {
-                                        if rv.is_empty() {
-                                            event.SetReturnValue(value);
-                                        }
-                                    }
-                                    None => {
-                                        event.upcast::<Event>().PreventDefault();
+                        if let Ok(value) = handler.Call_(object, event, exception_handle) {
```
Comment (Ms2ger) [inline components/script/dom/eventtarget.rs]: Can you add some tests for `returnValue`? Stick them in `event-handler-processing-algorithm.html` along with the others.

Comment (nox) [inline components/script/dom/eventtarget.rs]: What do you mean?

Comment (Ms2ger) [inline components/script/dom/eventtarget.rs]: We should have a test that uses `BeforeUnloadEvent` rather than `Event`.

**Comment by nox** [comment]
Seems like https://github.com/servo/servo/issues/8157 is back?

**Comment by jdm** [comment]
In a new timeout-based form, sure. We should file a new issue for that.

**Comment by pcwalton** [comment]
Over to @Ms2ger

**Comment by jdm** [comment]
We should probably postpone this until https://github.com/whatwg/html/issues/2297 is resolved.

**Comment by nox** [comment]
The HTML fix landed but the tests now require full unload support to run. We don't support that.

**Pull request closed without merging**